pkg/graph/builder.go

```diff
@@ -250,21 +250,40 @@ func (b *Builder) NewResourceGraphDefinition(originalCR *v1alpha1.ResourceGraphD
 	return resourceGraphDefinition, nil
 }
 
+func (b *Builder) buildExternalRefResource(externalRef *v1alpha1.ExternalRef, resourceObject *map[string]interface{}) {
```

Comment:
can you please add a comment stating what are the responsibilities of this function?

Reply:
also i think this function is just easier to read through if it was returning a `map[string]interface{}`

Reply:
done.